features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/TimelineItemReactionsView.kt

```diff
@@ -16,59 +16,180 @@
 
 package io.element.android.features.messages.impl.timeline.components
 
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.AddReaction
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.google.accompanist.flowlayout.FlowMainAxisAlignment
 import com.google.accompanist.flowlayout.FlowRow
+import io.element.android.features.messages.impl.R
+import io.element.android.features.messages.impl.timeline.aTimelineItemReactions
+import io.element.android.features.messages.impl.timeline.model.AggregatedReaction
 import io.element.android.features.messages.impl.timeline.model.TimelineItemReactions
-import io.element.android.features.messages.impl.timeline.model.aTimelineItemReactions
-import io.element.android.libraries.designsystem.preview.ElementPreviewDark
-import io.element.android.libraries.designsystem.preview.ElementPreviewLight
+import io.element.android.libraries.designsystem.preview.DayNightPreviews
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toPersistentList
+
+/**
+ * The maximum number of items that can be displayed before some items will be hidden
+ *
+ * TODO The threshold should be based on the number of rows, rather than items.
+ *       Once items would spill onto a third row, they should be hidden.
+ *       Note this could be particularly worthwhile to handle reactions that are
+ *       longer than a single character (as annotation keys are free text).
+ */
+private const val COLLAPSE_ITEMS_THRESHOLD = 8
 
 @Composable
-fun TimelineItemReactionsView(
+fun TimelineItemReactions(
     reactionsState: TimelineItemReactions,
     mainAxisAlignment: FlowMainAxisAlignment,
     onReactionClicked: (emoji: String) -> Unit,
     onMoreReactionsClicked: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
+    var expanded: Boolean by remember { mutableStateOf(false) }
```

Comment:
I think you'll loose the state if you scroll as the item is removed from the composition. If you really want to handle it in the Compose view, you need to use `rememberSaveable` instead.

Reply:
https://github.com/vector-im/element-x-android/pull/806/commits/923f41cadebfd0c7bf26e6f0d1f0aa602c52c042